### ISSUE TITLE
Dask pipelines fit method seems to be broken

### ISSUE THREAD

**Issue by wd15** [materialsinnovation/pymks#474]
Sklearn pipelines that use Dask ML objects can't take `y_data` that's chunked. Look at the following,

```
import warnings

import dask.array as da
from sklearn.pipeline import Pipeline
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import PolynomialFeatures
from sklearn.model_selection import train_test_split
from sklearn.model_selection import GridSearchCV
from dask_ml.decomposition import PCA
from dask_ml.preprocessing import PolynomialFeatures
from dask_ml.linear_model import LinearRegression as LinearRegressionDA
from sklearn.linear_model import LinearRegression as LinearRegressionSK
import matplotlib.pyplot as plt
import numpy as np

from pymks.fmks.data.multiphase import generate
from pymks.fmks.plot import plot_microstructures
from pymks.fmks.bases.primitive import PrimitiveTransformer
from pymks.fmks.correlations import TwoPointCorrelation, FlattenTransformer

warnings.filterwarnings('ignore')

da.random.seed(10)
np.random.seed(10)

tmp = [
    generate(shape=(50, 21, 21), grain_size=x, volume_fraction=(0.5, 0.5), chunks=50, percent_variance=0.15)
    for x in [(15, 2), (2, 15), (7, 7), (8, 3), (3, 9), (2, 2)]
]
x_data_gen = da.concatenate(tmp)


from pymks.fmks.data.elastic_fe import solve

y_stress = solve(x_data_gen, elastic_modulus=(310, 200), poissons_ratio=(0.28, 0.3), macro_strain=0.001)['stress']

y_data_gen = da.average(y_stress.reshape(y_stress.shape[0], -1), axis=1)


x_data = x_data_gen.persist()
y_data = y_data_gen.persist()


pipeline = Pipeline(
    steps=[
        ("discritize", PrimitiveTransformer(n_state=2, min_=0.0, max_=1.0)),
        ("correlations", TwoPointCorrelation(periodic_boundary=True, cutoff=21, correlations=[(1,1),(0,0)])),
        ("flatten", FlattenTransformer()),
        ("reducer", PCA(n_components=5)),
        ("poly", PolynomialFeatures(degree=2)),
        ("connector", LinearRegressionDA(solver_kwargs={"normalize":False})),
#       ("connector", LinearRegressionSK())
    ]
)



print(x_data.shape)
print(y_data.shape)

y_pred = pipeline.fit(x_data, np.array(y_data)).predict(x_data)

print(y_pred.chunks)
```

This requires that the `y_data` is changed to a numpy array to work. Is this correct? 

Moreover all the objects need to be Dask ML objects to actually return a Dask array at the end and, thus, to be parallel. All the Sklearn objects take a dask array but always return numpy arrays.. Anyway this bascially works now other than this issue with requiring a numpy array. `y_pred` is returned as a dask array with the correct chunks and the graph looks good.

I'm wondering if it's something to do with the fact that when doing any ML it's often required to have all the y_data used for fitting across all the chunks in general. Any thoughts?

**Comment by beyucel** [comment]
This issue was addressed in  #448

**Comment by wd15** [comment]
No longer believe that this is an issue
